### PR TITLE
refactor(Client): remove fetchAllMembers option

### PR DESCRIPTION
**Pull request by MattIPv4**
**Please describe the changes this PR makes and why it should be merged:**

Removes the `fetchAllMembers` from the client options, per https://github.com/discordjs/discord.js/discussions/5246, https://github.com/discordjs/discord.js/issues/5123#issuecomment-766329225, etc.

Updates the `random.js` test to have a version of the old internal logic for `fetchAllMembers`, for folks that really need it.

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Review by kyranet — APPROVED** [review]

File: test/random.js [hunk 1]
```diff
@@ -9,14 +9,30 @@ const Discord = require('../src');
...
+    console.log(`Failed to fetch all members before ready! ${err}\n${err.stack}`);
+  }
+
+  // Fetch all members in a new guild
+  client.on('guildCreate', guild => guild.members.fetch()
+    .catch(err => console.log(`Failed to fetch all members: ${err}\n${err.stack}`)));
+
+  // Fetch all members in a newly available guild
+  client.on('guildUpdate', (oldGuild, newGuild) => !oldGuild.available && newGuild.available ? guild.members.fetch()
+    .catch(err => console.log(`Failed to fetch all members: ${err}\n${err.stack}`)) : Promise.resolve());
```
Comment (vladfrangu) [inline test/random.js:34]: Should go outside ready

**Review by vladfrangu — CHANGES_REQUESTED** [review]
Nitpick and also consider fixing the annotations from actions

**Review by vladfrangu — APPROVED** [review]
LGTM, make sure all uses of `checkShardsReady` don't call await. you can also remove the NEARLY status, unless it's used somewhere

File: src/client/websocket/WebSocketManager.js [hunk 2]
```diff
@@ -391,27 +391,12 @@ class WebSocketManager extends EventEmitter {
    * Checks whether the client is ready to be marked as ready.
    * @private
    */
-  async checkShardsReady() {
+  checkShardsReady() {
```
Comment (NotSugden) [inline src/client/websocket/WebSocketManager.js:394]: typings for this method need changing as it no longer returns a Promise

**Review by NotSugden — CHANGES_REQUESTED** [review]

**Comment by MattIPv4** [comment]
https://github.com/discordjs/discord.js/blob/9ffcd83027f0fc06d69df21475865ad55138de01/src/client/websocket/WebSocketShard.js#L268 is the only remaining use of NEARLY.

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**